swap/swap.go

```diff
@@ -374,6 +397,42 @@ func (s *Swap) Deploy(ctx context.Context, backend swap.Backend, path string) er
 	return s.deploy(ctx, backend, path)
 }
 
+func (s *Swap) verifyContract(ctx context.Context, address common.Address) error {
+	swap, err := swap.InstanceAt(address, s.backend)
+
+	if err != nil {
+		return err
+	}
+
+	valid, err := swap.ValidateCode(ctx, s.backend, address)
```

Comment:
What do you think, wouldn't it be nicer if `ValidateCode` just returned an error (and returned `ErrNotASwapContract` right away)? So we would not have to check for `valid` here and we could just only do
```
if err !=nil {
    return err
}
```

Reply:
Will do that, just didn't want to change the signature from `ValidateCode` because it was also called at another unrelated part of the code.